Smoketest fails
```
DEBUG - Running Testcase: /usr/libexec/vyos/tests/smoke/cli/test_container.py
DEBUG - test_api_socket (__main__.TestContainer.test_api_socket) ... ok
DEBUG - test_basic (__main__.TestContainer.test_basic) ... ok
DEBUG - test_cpu_limit (__main__.TestContainer.test_cpu_limit) ... ok
DEBUG - test_dual_stack_network (__main__.TestContainer.test_dual_stack_network) ... ok
DEBUG - test_ipv4_network (__main__.TestContainer.test_ipv4_network) ... ok
DEBUG - test_ipv6_network (__main__.TestContainer.test_ipv6_network) ... ok
DEBUG - test_name_server (__main__.TestContainer.test_name_server) ... ERROR
DEBUG - test_name_server (__main__.TestContainer.test_name_server) ... FAIL
DEBUG - test_network_mtu (__main__.TestContainer.test_network_mtu) ... FAIL
DEBUG - test_no_name_server (__main__.TestContainer.test_no_name_server) ... FAIL
DEBUG - test_uid_gid (__main__.TestContainer.test_uid_gid) ... FAIL
DEBUG - 
DEBUG - ======================================================================
DEBUG - ERROR: test_name_server (__main__.TestContainer.test_name_server)
DEBUG - ----------------------------------------------------------------------
DEBUG - Traceback (most recent call last):
DEBUG -   File "/usr/libexec/vyos/tests/smoke/cli/test_container.py", line 106, in test_name_server
DEBUG -     self.cli_commit()
DEBUG -   File "/usr/libexec/vyos/tests/smoke/cli/base_vyostest_shim.py", line 86, in cli_commit
DEBUG -     self._session.commit()
DEBUG -   File "/usr/lib/python3/dist-packages/vyos/configsession.py", line 264, in commit
DEBUG -     out = self.__run_command([COMMIT])
DEBUG -           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
DEBUG -   File "/usr/lib/python3/dist-packages/vyos/configsession.py", line 201, in __run_command
DEBUG -     raise ConfigSessionError(output)
DEBUG - vyos.configsession.ConfigSessionError: [ container ]
DEBUG - Traceback (most recent call last):
DEBUG -   File "/usr/libexec/vyos/services/vyos-configd", line 137, in run_script
DEBUG -     script.apply(c)
DEBUG -   File "/usr/libexec/vyos//conf_mode/container.py", line 505, in apply
DEBUG -     cmd(f'systemctl restart vyos-container-{name}.service')
DEBUG -   File "/usr/lib/python3/dist-packages/vyos/utils/process.py", line 155, in cmd
DEBUG -     raise OSError(code, feedback)
DEBUG - PermissionError: [Errno 1] failed to run command: systemctl restart vyos-container-dns-test.service
DEBUG - returned:
DEBUG - exit code: 1
DEBUG - 
DEBUG - [[container]] failed
DEBUG - Commit failed
DEBUG - 
DEBUG - 
DEBUG - ======================================================================
DEBUG - FAIL: test_name_server (__main__.TestContainer.test_name_server)
DEBUG - ----------------------------------------------------------------------
DEBUG - Traceback (most recent call last):
DEBUG -   File "/usr/libexec/vyos/tests/smoke/cli/test_container.py", line 71, in tearDown
DEBUG -     self.assertEqual(units, [])
DEBUG - AssertionError: Lists differ: ['/run/systemd/system/vyos-container-dns-test.service'] != []
DEBUG - 
DEBUG - First list contains 1 additional elements.
DEBUG - First extra element 0:
DEBUG - '/run/systemd/system/vyos-container-dns-test.service'
DEBUG - 
DEBUG - - ['/run/systemd/system/vyos-container-dns-test.service']
DEBUG - + []
DEBUG - 
DEBUG - ======================================================================
DEBUG - FAIL: test_network_mtu (__main__.TestContainer.test_network_mtu)
DEBUG - ----------------------------------------------------------------------
DEBUG - Traceback (most recent call last):
DEBUG -   File "/usr/libexec/vyos/tests/smoke/cli/test_container.py", line 71, in tearDown
DEBUG -     self.assertEqual(units, [])
DEBUG - AssertionError: Lists differ: ['/run/systemd/system/vyos-container-dns-test.service'] != []
DEBUG - 
DEBUG - First list contains 1 additional elements.
DEBUG - First extra element 0:
DEBUG - '/run/systemd/system/vyos-container-dns-test.service'
DEBUG - 
DEBUG - - ['/run/systemd/system/vyos-container-dns-test.service']
DEBUG - + []
DEBUG - 
DEBUG - ======================================================================
DEBUG - FAIL: test_no_name_server (__main__.TestContainer.test_no_name_server)
DEBUG - ----------------------------------------------------------------------
DEBUG - Traceback (most recent call last):
DEBUG -   File "/usr/libexec/vyos/tests/smoke/cli/test_container.py", line 71, in tearDown
DEBUG -     self.assertEqual(units, [])
DEBUG - AssertionError: Lists differ: ['/run/systemd/system/vyos-container-dns-test.service'] != []
DEBUG - 
DEBUG - First list contains 1 additional elements.
DEBUG - First extra element 0:
DEBUG - '/run/systemd/system/vyos-container-dns-test.service'
DEBUG - 
DEBUG - - ['/run/systemd/system/vyos-container-dns-test.service']
DEBUG - + []
DEBUG - 
DEBUG - ======================================================================
DEBUG - FAIL: test_uid_gid (__main__.TestContainer.test_uid_gid)
DEBUG - ----------------------------------------------------------------------
DEBUG - Traceback (most recent call last):
DEBUG -   File "/usr/libexec/vyos/tests/smoke/cli/test_container.py", line 71, in tearDown
DEBUG -     self.assertEqual(units, [])
DEBUG - AssertionError: Lists differ: ['/run/systemd/system/vyos-container-dns-test.service'] != []
DEBUG - 
DEBUG - First list contains 1 additional elements.
DEBUG - First extra element 0:
DEBUG - '/run/systemd/system/vyos-container-dns-test.service'
DEBUG - 
DEBUG - - ['/run/systemd/system/vyos-container-dns-test.service']
DEBUG - + []
DEBUG - 
DEBUG - ----------------------------------------------------------------------
DEBUG - Ran 10 tests in 149.984s
```

--- a/interface-definitions/container.xml.in
+++ b/interface-definitions/container.xml.in
@@ -275,6 +275,7 @@
             </properties>
             <defaultValue>64</defaultValue>
           </leafNode>
+          #include <include/name-server-ipv4-ipv6.xml.i>
           <tagNode name="network">
             <properties>
               <help>Attach user defined network to container</help>

--- a/smoketest/scripts/cli/test_container.py
+++ b/smoketest/scripts/cli/test_container.py
@@ -108,6 +108,9 @@ class TestContainer(VyOSUnitTestSHIM.TestCase):
         n = cmd_to_json(f'sudo podman inspect {cont_name}')
         self.assertEqual(n['HostConfig']['Dns'][0], name_server)
 
+        tmp = cmd(f'sudo podman exec -it {cont_name} cat /etc/resolv.conf')
+        self.assertIn(name_server, tmp)
+
     def test_cpu_limit(self):
         cont_name = 'c2'
 

--- a/smoketest/scripts/cli/test_container.py
+++ b/smoketest/scripts/cli/test_container.py
@@ -96,6 +96,18 @@ class TestContainer(VyOSUnitTestSHIM.TestCase):
         tmp = cmd(f'sudo podman exec -it {cont_name} sysctl kernel.msgmax')
         self.assertEqual(tmp, 'kernel.msgmax = 4096')
 
+    def test_name_server(self):
+        cont_name = 'dns-test'
+        name_server = '192.168.0.1'
+        self.cli_set(base_path + ['name', cont_name, 'allow-host-networks'])
+        self.cli_set(base_path + ['name', cont_name, 'image', cont_image])
+        self.cli_set(base_path + ['name', cont_name, 'name-server', name_server])
+
+        self.cli_commit()
+
+        n = cmd_to_json(f'sudo podman inspect {cont_name}')
+        self.assertEqual(n['HostConfig']['Dns'][0], name_server)
+
     def test_cpu_limit(self):
         cont_name = 'c2'
 

--- a/src/conf_mode/container.py
+++ b/src/conf_mode/container.py
@@ -363,9 +363,14 @@ def generate_run_arguments(name, container_config):
     if 'allow_host_pid' in container_config:
       host_pid = '--pid host'
 
+    name_server = ''
+    if 'name_server' in container_config:
+        for ns in container_config['name_server']:
+            name_server += f'--dns {ns}'
+
     container_base_cmd = f'--detach --interactive --tty --replace {capabilities} --cpus {cpu_quota} {sysctl_opt} ' \
                          f'--memory {memory}m --shm-size {shared_memory}m --memory-swap 0 --restart {restart} ' \
-                         f'--name {name} {hostname} {device} {port} {volume} {env_opt} {label} {uid} {host_pid}'
+                         f'--name {name} {hostname} {device} {port} {name_server} {volume} {env_opt} {label} {uid} {host_pid}'
 
     entrypoint = ''
     if 'entrypoint' in container_config:

